src/status_im2/contexts/wallet/events.cljs

```diff
@@ -75,3 +77,71 @@
                            data-store/<-rpc)
                      data)}]
      {:db (assoc db :wallet/networks network-data)})))
+
+(def collectibles-request-batch-size 1000)
+
+(defn- displayable-collectible?
+  [{:keys [image_url animation_url]}]
+  (or (not= "" animation_url)
+      (not= "" image_url)))
+
+(rf/reg-event-fx :wallet/store-collectibles
+ (fn [{:keys [db]} [collectibles]]
+   (let [stored-collectibles      (get-in db [:wallet :collectibles])
+         displayable-collectibles (filter displayable-collectible? collectibles)]
+     {:db (assoc-in db
+           [:wallet :collectibles]
+           (reduce conj displayable-collectibles stored-collectibles))})))
+
+(rf/reg-event-fx :wallet/clear-stored-collectibles
+ (fn [{:keys [db]}]
+   {:db (assoc-in db [:wallet :collectibles] [])}))
+
+(rf/reg-event-fx :wallet/save-collectibles-request-details
+ (fn [{:keys [db]} [request-details]]
+   {:db (assoc-in db [:wallet :ongoing-collectibles-request] request-details)}))
+
+(rf/reg-event-fx :wallet/clear-collectibles-request-details
+ (fn [{:keys [db]}]
+   {:db (assoc-in db [:wallet :ongoing-collectibles-request] {})}))
+
+(rf/reg-event-fx :wallet/request-collectibles
+ (fn [{:keys [db]} [{:keys [addresses offset new-request?]}]]
+   (let [request-params [0
+                         [(chain/chain-id db)]
+                         addresses
+                         offset
+                         collectibles-request-batch-size]]
+     (merge
+      {:json-rpc/call [{:method     "wallet_filterOwnedCollectiblesAsync"
+                        :params     request-params
+                        :on-success #()
+                        :on-error   (fn [error]
+                                      (log/error "failed to request collectibles"
+                                                 {:event  :wallet/request-collectibles
+                                                  :error  error
+                                                  :params request-params})
+                                    )}]}
+      (when new-request?
+        {:fx
+         [[:dispatch [:wallet/clear-stored-collectibles]]
+          [:dispatch [:wallet/save-collectibles-request-details {:addresses addresses}]]]})))))
```

Comment:
Cheaper than `merge` two maps with only one key each, is using `assoc`. But in this case I think we could avoid using `merge`.

The vector passed to `:fx` can contain `nil`, but I'm not sure if it can be `nil`. So:
```clojure
{:json-rpc/call [{:method     "wallet_filterOwnedCollectiblesAsync"
                  :params     request-params
                  :on-success #()
                  :on-error   (fn [error]
                                (log/error "failed to request collectibles"
                                           {:event  :wallet/request-collectibles
                                            :error  error
                                            :params request-params}))}]
 :fx            (when new-request?
                  [[:dispatch [:wallet/clear-stored-collectibles]]
                   [:dispatch [:wallet/save-collectibles-request-details {:addresses addresses}]]])}
```

if that doesn't work, we can change `(when new-request? [...])` by `(if new-request? [...] [])`